blockchain/filedao/filedao.go

```diff
@@ -98,19 +129,19 @@ func (fd *fileDAO) Height() (uint64, error) {
 }
 
 func (fd *fileDAO) GetBlockHash(height uint64) (hash.Hash256, error) {
-	var (
-		h   hash.Hash256
-		err error
-	)
-	for _, file := range fd.v2Fd {
-		if h, err = file.GetBlockHash(height); err == nil {
-			return h, nil
+	if fd.v2Fd != nil {
+		if height == 0 {
+			return hash.ZeroHash256, nil
+		}
+		if v2 := fd.v2Fd.FileDAOByHeight(height); v2 != nil {
```

Comment:
per Yutong suggestion,  use the start height to find out which DB owns the height, no need to run a for {} loop and try every file